web/src/message_list_view.ts

```diff
@@ -425,6 +427,14 @@ function maybe_restore_focus_to_message_edit_form(): void {
     }, 0);
 }
 
+function is_search_view(): boolean {
+    const current_filter = narrow_state.filter();
+    if (current_filter?.sorted_term_types().includes("search")) {
+        return true;
+    }
+    return false;
+}
```

Comment:
I think you want to use `supports_collapsing_recipients` instead, since views like "starred messages" are also search views.